platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/FilterInjector.java

```diff
@@ -173,4 +187,45 @@ public void injectFilter(ServiceReference<ServletContext> serviceReference) {
         }
     }
 
+    public void addListener(ServiceReference<HttpSessionListener> sessionListener) {
+        if (sessionListener != null) {
+            Bundle refBundle = sessionListener.getBundle();
+            if (refBundle != null) {
+                BundleContext bundlectx = refBundle.getBundleContext();
+                if (bundlectx != null) {
+                    sessionListeners.add(bundlectx.getService(sessionListener));
+                }
+            }
+        }
+    }
+
+    public void removeListener(ServiceReference<HttpSessionListener> sessionListener) {
+        if (sessionListener != null) {
+            Bundle refBundle = sessionListener.getBundle();
+            if (refBundle != null) {
+                BundleContext bundlectx = refBundle.getBundleContext();
+                if (bundlectx != null) {
+                    sessionListeners.remove(bundlectx.getService(sessionListener));
```

Comment:
Same comment about checking for a null here.

Which makes me think we might want to change `sessionListeners` to be a map. That way, you could simplify this remove method to simply pull the entry with this `sessionListener` key out. Changing up the `WrapperListener` would be a trivial change to look at the map's values.